packages/api/src/routes/mint.ts

```diff
@@ -92,29 +92,32 @@ const mint: FastifyPluginAsync = async (fastify, opts): Promise<void> => {
       const abi = WITTY_BUFFICORNS_ERC721_ABI
 
       const contract = new web3.eth.Contract(abi, WITMON_ERC721_ADDRESS)
-      // TODO: ranch id goes from 0 to 5, ensure compatibility with solidity
       const ranchId = RANCH_TO_INDEX[player.ranch]
 
-      let callResult
-      try {
-        callResult = await contract.methods.getRanchWeather(ranchId).call()
-      } catch (err) {
-        console.error(err)
-        return reply
-          .status(404)
-          .send(new Error(`Error calling getRanchWeather method from contract`))
+      let ranchWeather = fastify.cache.getWeather(ranchId)
```

Comment:
Calling getRanchWeather() will always work, no matter if contract is ready for minting or not. Even if no weather description had been yet solved by Witnet, getRanchWeather() would return empty string, but it will never fail.

So, I'd rather suggest to call `getStatus()` at begining of the `mint` implementation, as to make sure the contract is ready for minting awards. A value of `2` means the contract is ready for minting, while `1` means a randomer number is being solved, and `0` no random number has yet been requested.